server/util/util.ts

```diff
@@ -112,3 +112,65 @@ export const timestampColumnTransformer: ValueTransformer = {
 };
 
 export const oneDaySeconds = 24 * 60 * 60;
+
+export type TimeInterval =
+  | 'day'
+  | 'week'
+  | 'month'
+  | 'year';
+
+export function getEsTimeInterval(interval: TimeInterval) {
+  switch (interval) {
+    case 'day': return 'd';
+    case 'week': return 'w';
+    case 'month': return 'M';
+    case 'year': return 'y';
+    default:
+      throw new Error(`Unsupported time interval '${interval}'`);
+  }
+}
+
+export function getEsDateFormat(interval: TimeInterval) {
+  switch (interval) {
+    case 'day':
+    case 'week': return 'yyyy-MM-dd';
+    case 'month': return 'yyyy-MM';
+    case 'year': return 'yyyy';
+    default:
+      throw new Error(`Unsupported interval '${interval}'`);
+  }
+}
+
+export function getMomentDateFormat(interval: TimeInterval) {
+  switch (interval) {
+    case 'day':
+    case 'week': return 'yyyy-MM-DD';
+    case 'month': return 'yyyy-MM';
+    case 'year': return 'yyyy';
+    default:
+      throw new Error(`Unsupported interval '${interval}'`);
+  }
+}
+
+export function requireQuery<TQuery extends object>(
```

Comment:
Super knit-picky but this function name is confusing to me. I think I was thinking of the `require` as `import` - since we're in the JS world. What do you think about ensureRequiredFields or validateQueryKeys or verify/assert 
 ... Something